ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/RecursivePeerRefreshState.java

```diff
@@ -33,12 +33,13 @@
   private final BondingAgent bondingAgent;
   private final NeighborFinder neighborFinder;
   private final List<PeerDistance> anteList;
-  private final List<OutstandingRequest> outstandingRequestList;
+
+  private final List<OutstandingRequest> outstandingBondingRequestList;
+  private final List<OutstandingRequest> outstandingNeighboursRequestList;
+
   private final LinkedHashSet<BytesValue> dispatchedFindNeighbours;
   private final LinkedHashSet<BytesValue> dispatchedBond;
 
-  private List<DiscoveryPeer> bootstrapPeers;
-
   RecursivePeerRefreshState(
```

Comment:
Adding some notes based on our offline discussion:

We talked about the need to define a terminating condition for this recursive peer search.  If we have a defined terminating condition, this allows us to issue a new request for peers when a peer refresh is requested.  If the previous search has concluded, we can start a new search.  Otherwise, we should wait for the previous search to conclude before issuing a new search.  

The docs [here](https://github.com/ethereum/devp2p/blob/master/discv4.md#recursive-lookup) define that terminating condition:
> The lookup terminates when the initiator has queried and gotten responses from the k closest nodes it has seen.